### PR TITLE
Use contenteditable as opposed to textarea for text fields

### LINKED ISSUE #345: Form ignores settings for verticalAlignment
### Describe the bug

When a field has `verticalAlignment` set to `middle` for example, the text is not centered vertically when the template is loaded as `Form`. It renderes correctly in `Designer`, `Viewer` and when a PDF is generated.

### To Reproduce

1. Start the Playground
2. Set the vertical alignment to `middle`
3. Save the template and switch to the `Form` view

### Expected behavior

Apply the same settings as in `Designer` and `Viewer`.

### Your Environment

```markdown
- pdfme version: `main` branch
> $ git log -n 1 --pretty=format:"%H"
> 3258e441b18c7c93cdf2a715a904213d9b97fc6c
- Operating system: Ubuntu 23.10
- Node.js version or Browser name & version: Node v20.10, Chrome 119.0
```

### Your Error Log

```shell
There is no error log, but here is a screencast below:
```

### Additional context

[Screencast from 2023-12-01 23-04-24.webm](https://github.com/pdfme/pdfme/assets/227762/324403c2-b079-45ee-b715-5be44dd551a7)

### PR DESCRIPTION
**Pull request by peteward**
fixes #345 - for text fields to respect vertical alignment while in editing mode.

Another positive point about using contenteditable is that the style is the same for when you are in edit mode and when you are not. You do not have to worry about 'adjusting' font positions in the different cases.

However, I think we might have lost some positioning adjustments with the custom schema work, so I will revisit that in a separate issue.

Example of change:

https://github.com/pdfme/pdfme/assets/7068515/cce219e5-23c2-4562-89a0-0dba8b51fe74

### CODE REVIEW

File: packages/schemas/src/text/uiRender.ts [hunk 1]
```diff
@@ -119,45 +120,56 @@ export const uiRender = async (arg: UIRenderProps<TextSchema>) => {
...
+      const placeholderStyle = document.createElement('style');
+      placeholderStyle.textContent = `
+        [contenteditable=true]:empty:before {
+          content: attr(placeholder);
+          pointer-events: none;
+          display: block;
+        }
+      `;
+      container.appendChild(placeholderStyle);
+    }
```
Comment (peteward) [inline packages/schemas/src/text/uiRender.ts:156]: There was an implementation for `placeholder` before, so I have added this to maintain the behaviour, although I'm not sure if it's possible to use it?

**Comment by peteward** [comment]
hmm tests all pass locally which is strange.. 
```
> @pdfme/ui@0.0.0 test
> jest

 PASS  __tests__/helper.test.ts
 PASS  __tests__/components/Designer.test.tsx
 PASS  __tests__/components/Preview.test.tsx

Test Suites: 3 passed, 3 total
Tests:       13 passed, 13 total
Snapshots:   3 passed, 3 total
Time:        4.143 s
Ran all test suites.
```

I will check again tomorrow

File: packages/schemas/src/text/uiRender.ts (hunk 1 above)
Comment (hand-dot) [reply packages/schemas/src/text/uiRender.ts:156]: I checked, but the placeholder disappears. We need to think of a way to display the placeholder in the div.

**Comment by hand-dot** [comment]
Hey @peteward 

Can you check bellow changes?

- [Fix Dynamic font sizing no longer truly dynamic #357
](https://github.com/pdfme/pdfme/pull/359/commits/88868c95028bfdc2e5478a9a55f7aed8b3b66157)
- [Fix placeholder](https://github.com/pdfme/pdfme/pull/359/commits/179a895afc30d5ae3ef46219a5ded2e8ec072515)

https://github.com/pdfme/pdfme/assets/24843808/4ae5ffef-d18a-4ef4-b057-94fac8c857dc

File: packages/schemas/src/text/uiRender.ts [hunk 2]
```diff
@@ -118,45 +122,66 @@ export const uiRender = async (arg: UIRenderProps<TextSchema>) => {
...
-    container.appendChild(textarea);
+    textBlock.contentEditable = 'plaintext-only';
+    textBlock.tabIndex = tabIndex || 0;
+    textBlock.innerText = value;
+    textBlock.addEventListener('blur', (e: Event) => {
+      onChange && onChange((e.target as HTMLDivElement).innerText);
+      stopEditing && stopEditing();
+    });
+
+    textBlock.addEventListener('keypress', () => {
```
Comment (peteward) [inline packages/schemas/src/text/uiRender.ts:145]: ah of course, you can just do it this way 🤦  

I will test this locally very soon

**Comment by peteward** [comment]
This is a great improvement. A couple of issues to be looked at though:

1. The topAdj and bottomAdj need to be recalculated when font size changes, so need to also be called on the keyUp handler.
2. It's still a lot jankier than it used to be.

I'm happy to look into both of these issues over the weekend.

**Comment by peteward** [comment]
Updated:
* use `keyup` instead of `keypress` so that it triggers when backspace and delete are pressed
* recalc the dynamic font size based on the previous calculated font size (rather than the base one) as this considerably reduces the number of calculation iterations required.
* recalculate the bottom and top adjustments along with font size.

@hand-dot I think this one is good to go now! Can you confirm?

**Comment by hand-dot** [comment]
@peteward 
Cool! I'll check it and If it okay i gonna merge this PR.

Thank you

**Review by hand-dot — APPROVED** [review]
LGTM!
Really nice!!

**Comment by hand-dot** [comment]
Released V3.1.4

**Pull request merged**